src/main/resources/lib/portal.ts

```diff
@@ -1,16 +1,70 @@
-import { query, Site } from "/lib/xp/content";
-import { forceArray } from "./utils";
+import { query, type Site, get } from "/lib/xp/content";
+import { forceArray, buildBaseContext } from "./utils";
+import { list, type Repository } from "/lib/xp/repo";
+import { run } from "/lib/xp/context";
+import type { SiteConfig } from "/site/index";
+import { EnonicEventDataNode } from "/lib/xp/event";
 
-export function getSiteInCron<Config extends object>(): Site<Config> {
+export interface RepoSiteAppConfig {
+  repoId: string,
+  siteName: string,
+  appConfig: SiteConfig,
+}
+
+function getSites<Config extends object>(): Site<Config>[] {
   return query<Site<Config>>({
-    count: 1,
     query: `_path LIKE '/content/*' AND data.siteConfig.applicationKey = '${app.name}'`,
-    contentTypes: ["portal:site"],
-  }).hits[0];
+    contentTypes: ["portal:site"]
+  }).hits;
+}
+
+const runInDraftRepoContext = (callback: () => void, repositoryId: string, params: []) => {
+  if (!repositoryId) {
+    repositoryId = "com.enonic.cms.default"; // Best guess... This shouldn't happen!
+  }
+
+  return run(buildBaseContext(repositoryId), () => callback(...params));
+};
+
+export function getAllSiteConfigsInCron(): RepoSiteAppConfig[] {
+  const siteConfigs: RepoSiteAppConfig[] = [];
+  const repolist: Repository[] = list();
```

Comment:
I don't think this `const` is necessary, just use `list()` / `listRepositories()` directly on the next line.